tests/subsys/net/lib/wifi_credentials/src/main.c

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <unity.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/fff.h>
+
+#include <net/wifi_credentials.h>
+
+#include "wifi_credentials_internal.h"
+
+DEFINE_FFF_GLOBALS;
+
+FAKE_VALUE_FUNC(int, wifi_credentials_store_entry, size_t, const void *, size_t)
+FAKE_VALUE_FUNC(int, wifi_credentials_load_entry, size_t, void *, size_t)
+FAKE_VALUE_FUNC(int, wifi_credentials_delete_entry, size_t)
+
+uint8_t fake_settings_buf[CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES][ENTRY_MAX_LEN];
+
+int custom_wifi_credentials_store_entry(size_t idx, const void *buf, size_t buf_len)
+{
+	TEST_ASSERT(idx < CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES);
+	memcpy(fake_settings_buf[idx], buf, MIN(ENTRY_MAX_LEN, buf_len));
+	return 0;
+}
+
+int custom_wifi_credentials_load_entry(size_t idx, void *buf, size_t buf_len)
+{
+	TEST_ASSERT(idx < CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES);
+	memcpy(buf, fake_settings_buf[idx], MIN(ENTRY_MAX_LEN, buf_len));
+	return 0;
+}
+
+void setUp(void)
+{
+	RESET_FAKE(wifi_credentials_store_entry);
+	RESET_FAKE(wifi_credentials_load_entry);
+	RESET_FAKE(wifi_credentials_delete_entry);
+	wifi_credentials_store_entry_fake.custom_fake =
+		custom_wifi_credentials_store_entry;
+	wifi_credentials_load_entry_fake.custom_fake =
+		custom_wifi_credentials_load_entry;
+}
+
+#define SSID1 "test1"
+#define PSK1 "super secret"
+#define SECURITY1 WIFI_SECURITY_TYPE_PSK
+#define BSSID1 "abcdef"
+#define FLAGS1 WIFI_CREDENTIALS_FLAG_BSSID
+
+#define SSID2 "test2"
+#define PSK2 NULL
+#define SECURITY2 WIFI_SECURITY_TYPE_NONE
+#define BSSID2 NULL
+#define FLAGS2 0
+
+#define SSID3 "test3"
+#define PSK3 "extremely secret"
+#define SECURITY3 WIFI_SECURITY_TYPE_SAE
+#define BSSID3 NULL
+#define FLAGS3 0
```

Comment:
I think your code now supports n-length all-null SSIDs, right?

I recommend writing a test or two involving those

Also maybe a test with `\0` as a leading character for the SSID

Reply:
This now supports having one (1) zero-length SSID network registered. I'll add some tests. Do you think this has to be documented?

Reply:
I think the length limitations and support of `\0` in the SSID should be documented, yeah

But I wasn't referring to zero-length, I was referring to non-zero-length SSID containing one or more instance of the character `\0`